toponetx/algorithms/distance.py

```diff
@@ -0,0 +1,144 @@
+"""Module to distance between nodes or cells on topological domains."""
+from collections.abc import Collection, Hashable, Iterable, Iterator
+from warnings import warn
+
+import networkx as nx
+import numpy as np
+
+from toponetx.classes.cell import Cell
+from toponetx.classes.cell_complex import CellComplex
+from toponetx.classes.colored_hypergraph import ColoredHyperGraph
+from toponetx.classes.combinatorial_complex import CombinatorialComplex
+from toponetx.classes.hyperedge import HyperEdge
+from toponetx.classes.simplicial_complex import SimplicialComplex
+from toponetx.exception import TopoNetXError
+
+__all__ = ["distance", "cell_distance"]
+
+
+def distance(cmplex, source, target, s=1):
+    """Return shortest s-walk distance between two nodes in the cell complex.
+
+    Parameters
+    ----------
+    source : node.uid or node
+        a node in the cell complex
+    target : node.uid or node
+        a node in the cell complex
+    s : int
+        the number of cells
+
+    Returns
+    -------
+    s_walk_distance : int
+
+    See Also
+    --------
+    cell_distance
+
+    Notes
+    -----
+    The s-distance is the shortest s-walk length between the nodes.
+    An s-walk between nodes is a sequence of nodes that pairwise share
+    at least s cells. The length of the shortest s-walk is 1 less than
+    the number of nodes in the path sequence.
+
+    Uses the networkx shortest_path_length method on the graph
+    generated by the s-adjacency matrix.
+
+    Example
+    -------
+    >>> CC = CellComplex()
+    >>> CC.add_cell([2,3,4],rank=2)
+    >>> CC.add_cell([5,6,7],rank=2)
+    >>> list(node_diameters(CC))
+    >>> CCC = CC.to_combinatorial_complex()
+    >>> list(node_diameters(CCC))
+    >>> CHG = CC.to_colored_hypergraph()
+    >>> list(node_diameters(CHG))
+    """
+    if not isinstance(cmplex, (CellComplex, CombinatorialComplex, ColoredHyperGraph)):
+        raise ValueError("Input complex {cplex} is not supported.")
+    if isinstance(source, Cell):
+        source = source.elements
+    if isinstance(target, Cell):
+        target = target.elements
+    if isinstance(source, Iterable):
+        source = tuple(source)
+    if isinstance(target, Iterable):
+        target = tuple(target)
+    rowdict, A = cmplex.node_to_all_cell_adjacnecy_matrix(index=True)
+    G = nx.from_scipy_sparse_array(A)
+    try:
+        path = nx.shortest_path_length(G, rowdict[source], rowdict[target])
+        return path
+    except Exception:
+        warn(f"No {s}-path between {source} and {target}")
+        return np.inf
+
+
+def cell_distance(cmplex, source, target, s=1):
+    """Return the shortest s-walk distance between two cells in the cell complex.
+
+    Parameters
+    ----------
+    source : cell.uid or cell
+        a cell in the cell complex
```

Comment:
This does not respect our docstrings conventions, and - as a result - will not look clean on the doc website: update?
Cf: https://pyt-team.github.io/topomodelx/contributing/index.html#write-documentation

Reply:
done